src/vcdiff.cc

```diff
@@ -177,21 +178,20 @@ void VcdCtx::Process(const char* data, size_t len, bool is_last) {
 
 }
 
-bool VcdCtx::CheckError() {
+bool VcdCtx::CheckError(v8::Isolate* isolate) {
   if (!HasError())
     return true;
 
-  SendError();
+  SendError(isolate);
   return false;
 }
 
-void VcdCtx::SendError() {
-  v8::HandleScope scope;
+void VcdCtx::SendError(v8::Isolate* isolate) {
   v8::Local<v8::Value> args[2] = {
-    v8::String::New(GetErrorString(err_)),
-    v8::Number::New(static_cast<int>(err_)),
+    v8::String::NewFromUtf8(isolate, GetErrorString(err_)),
```

Comment:
I'm no expert, but from what I can tell of https://developers.google.com/v8/embed?hl=en#handles-and-garbage-collection and the source code ... because `v8::Local`'s ctor takes an `isolate` now.

Since Node 0.12, a `v8::Isolate` is basically a stack of `v8::HandleScope` objects. The HandleScope ctor adds a node, and the HandleScope dtor removes one. So `v8::Local`'s ctor is basically: "add a `Handle` to the top of `isolate`'s stack."

Back in Node 0.10, `v8::Isolate` didn't exist. `HandleScope` matched up one-to-one with the CPU stack.

The only guarantee we need is that somewhere in the `isolate`, there's a HandleScope. I gather, from reading various comments elsewhere, that best practice is to have a HandleScope per JS function. So little functions that make C++ more legible don't need them.

Disclaimer: I'm no expert.